instrumentation/messaging/src/main/java/brave/messaging/MessagingParser.java

```diff
@@ -0,0 +1,43 @@
+package brave.messaging;
+
+import brave.SpanCustomizer;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+
+public class MessagingParser {
+
+  public <Chan, Msg> void message(ChannelAdapter<Chan> channelAdapter,
+      MessageAdapter<Msg> messageAdapter,
+      Chan channel, Msg message, SpanCustomizer customizer) {
+    customizer.name(messageAdapter.operation(message));
+    channel(channelAdapter, channel, customizer);
+    identifier(messageAdapter, message, customizer);
+  }
+
+  public <Chan> void channel(ChannelAdapter<Chan> adapter, Chan chan,
+      SpanCustomizer customizer) {
+    String channel = adapter.channel(chan);
+    if (chan != null) customizer.tag(adapter.channelTagKey(chan), channel);
+  }
+
+  public <Msg> void identifier(MessageAdapter<Msg> adapter, Msg message,
+      SpanCustomizer customizer) {
+    String identifier = adapter.identifier(message);
+    if (identifier != null) {
+      customizer.tag(adapter.identifierTagKey(), adapter.identifier(message));
```

Comment:
Small nit but I think this could just be `customizer.tag(adapter.identifierTagKey(), identifier);`.  Also should we do a null and empty string check on `adapter.identifierTagKey()` just to ensure a bad adapter doesn't cause issues?

Reply:
```suggestion
      customizer.tag(adapter.identifierTagKey(), identifier);
```